### ISSUE TITLE
finding HEAD in a submodule

### ISSUE THREAD

**Issue by nilswiersma** [snoyberg/githash#16]
Hi,
a submodule contains a .git file, rather than a directory, with the line
`gitdir: <some path>/.git/modules/<name of module>`
where `<name of module>/` contains `.git/HEAD`.

`githash` is currently not aware of this and throws the following error:

```
    • Exception when trying to run compile-time code:
        GHECouldn'tReadFile "<some path>/.git/HEAD" <some path>/.git/HEAD: openBinaryFile: inappropriate type (Not a directory)
      Code: tGitInfoCwd
    • In the Template Haskell splice $$tGitInfoCwd
      In the first argument of ‘giHash’, namely ‘$$tGitInfoCwd’
      In the expression: giHash $$tGitInfoCwd
   |
35 | gitHash = giHash $$tGitInfoCwd
   |                  ^^^^^^^^^^^^^
```

Thanks!

**Comment by snoyberg** [comment]
Would you be interested in sending a PR?

**Comment by zachjs** [comment]
I believe this was due to the use of an outdated version of githash. @nilswiersma, can you see if this is now fixed, as mentioned in zachjs/sv2v#84?

**Comment by nilswiersma** [comment]
You are right, submodules works as expected now, thanks!
